Include/internal/pycore_bitutils.h

```diff
@@ -131,8 +131,43 @@ _Py_popcount32(uint32_t x)
 }
 
 
+// Return the index of the most significant 1 bit in 'x'. This is the smallest
+// integer k such that x < 2**k. Equivalent to floor(log2(x)) + 1 for x != 0.
+static inline int
+_Py_bit_length(unsigned long x)
+{
+    if (x == 0) {
+        return 0;
+    }
+#if (defined(__clang__) || defined(__GNUC__))
+    // __builtin_clzl() is available since GCC 3.4.
+    // Undefined behavior for x == 0.
+    return sizeof(unsigned long) * 8 - __builtin_clzl(x);
+#elif defined(_MSC_VER)
+    Py_BUILD_ASSERT(sizeof(unsigned long) <= 4);
+    // Does not write to msb if x == 0.
+    unsigned long msb;
+    _BitScanReverse(&msb, x);
+    return (int)msb + 1;
+#else
+    int msb = 0;
+    do {
+        msb += 1;
+        x >>= 1;
+    } while (x != 0);
+    return msb;
+#endif
+}
+
+static inline int
+_Py_bit_length_digit(digit x)
```

Comment:
I would prefer to only declare it in longobject.c as bit_length_digit(). I don't think that it's useful in any other file.